### ISSUE TITLE
Trigger modal from Controller

### ISSUE THREAD

**Issue by b-d-b** [ef4/liquid-fire#185]
Is it possible to trigger the modal without a link-to?  I have a condition that, if not met on page load, should open the modal.  However, I cannot seem to trigger this behavior outside of using a link-to.  Perhaps I am missing something?

Thanks for your work, the modal looks great and the transitions are extra crispy!

**Comment by ef4** [comment]
Yes, the modal is actually just triggered by setting the value of your `withParams` on your controller. Using link-to with query params is only one way to do that. You can also just set it directly from your route's `setupController` hook, for example. Or make your `withParams` point at a computed property that will decide on its own when the modal should show based on something else.

**Comment by b-d-b** [comment]
I am not sure if I understand correctly, but maybe you can help me if I share the example:

```
print_config = {}

this.application = this.get('application');
this.configure(this.application);
if(!this.print_config.print_location)
	this.open_print_config_in_modal();
```

Configure simply attempts to retrieve some settings from localStorage, and if those values are not present, we want to open a modal with a form component rendered inside of it, forcing the user to configure their printer at that point.

However, right now, we cannot seem to get this to work with our print configuration component, so I am trying to do something like this (pseudo code):

```
open_print_config_in_modal: function(){
    this.modal('print-config').open()
}
```

I am wondering if this is even possible?

**Comment by b-d-b** [comment]
This is all inside my print-config.js component file, mind you.  I call the top-most block on didInsertElement.

**Comment by jgodi** [comment]
Here is the way that I do it:

Router.js
```
Router.map(function() {
  this.route('myRoute', function() {
    this.modal('myModal', { withParams: ['myModalTrigger', 'myModalTrigger'] });
  });
});
```

myModal.controller.js
```this.set('myModalTrigger', true)```

The ```withParams``` watches for those properties to be non-default, so you can trigger them on your own, then reset them to hide the modal.

[Documentation Link](http://ef4.github.io/liquid-fire/#/modals), just instead of the ```{{#link-to}}``` just control that property in your controller.

Hope that helped.

**Comment by b-d-b** [comment]
Thanks, that did help.

**Comment by davidgovea** [comment]
I'm using `.transitionToRoute()` like so:  
```js
this.transitionToRoute({queryParams: {myModalTrigger: true}});
```

see [here](https://github.com/emberjs/ember.js/blob/v1.10.0/packages/ember-routing/lib/ext/controller.js#L217)

**Comment by whodies** [comment]
Hi guys, really awesome module. The documentation specifically states that triggering the modal with query params is optional. However, I haven't been able to trigger it without query params. Sometimes I need a modal to be separate from the url, for example if the modal serves as an alert or notification.

Thanks again

**Comment by ef4** [comment]
@whodies the modal just watches your controller to see if the params change. It doesn't even know about queryParams. Can't really help more without seeing an example of what you're trying.
